api/src/main/java/net/kyori/adventure/text/JoinConfiguration.java

```diff
@@ -0,0 +1,236 @@
+   * Gets the suffix of this join configuration.
+   *
+   * @return the suffix
+   * @since 4.8.0
+   */
+  @Nullable Component suffix();
+
+  /**
+   * Gets the separator of this join configuration.
+   *
+   * @return the separator
+   * @since 4.8.0
+   */
+  @Nullable Component separator();
+
+  /**
+   * Gets the last separator of this join configuration.
+   *
+   * @return the last separator
+   * @since 4.8.0
+   */
+  @Nullable Component lastSeparator();
+
+  /**
+   * Gets the last separator that will be used instead of the normal last separator in the case where there
+   * are more than two components being joined. This can be used to mimic a serial (or Oxford) comma.
+   *
+   * @return the separator
+   * @since 4.8.0
+   */
+  @Nullable Component lastSeparatorIfSerial();
+
+  /**
+   * Gets the operator of this join configuration.
+   *
+   * @return the operator
+   * @since 4.8.0
+   */
+  @NotNull UnaryOperator<Component> operator();
+
+  /**
+   * A builder for join configurations.
+   *
+   * @since 4.8.0
+   */
+  interface Builder extends Buildable.Builder<JoinConfiguration> {
+    /**
+     * Sets the prefix of this join configuration builder.
+     *
+     * @param prefix the prefix
+     * @return this builder
+     * @since 4.8.0
+     */
+    @Contract("_ -> this")
+    @NonNull Builder prefix(final @Nullable ComponentLike prefix);
+
+    /**
+     * Sets the suffix of this join configuration builder.
+     *
+     * @param suffix the suffix
+     * @return this builder
+     * @since 4.8.0
+     */
+    @Contract("_ -> this")
+    @NonNull Builder suffix(final @Nullable ComponentLike suffix);
+
+    /**
+     * Sets the separator of this join configuration builder.
+     *
+     * @param separator the separator
+     * @return this builder
+     * @since 4.8.0
+     */
+    @Contract("_ -> this")
+    @NonNull Builder separator(final @Nullable ComponentLike separator);
+
+    /**
+     * Sets the last separator of this join configuration builder.
+     *
+     * @param lastSeparator the last separator
+     * @return this builder
+     * @since 4.8.0
+     */
+    @Contract("_ -> this")
+    @NonNull Builder lastSeparator(final @Nullable ComponentLike lastSeparator);
+
+    /**
+     * Sets the last separator that will be used instead of the normal last separator in the case where there
+     * are more than two components being joined. This can be used to mimic a serial (or Oxford) comma.
+     *
+     * @param lastSeparatorIfSerial the last separator
+     * @return this builder
+     * @since 4.8.0
+     */
+    @Contract("_ -> this")
+    @NonNull Builder lastSeparatorIfSerial(final @Nullable ComponentLike lastSeparatorIfSerial);
+
+    /**
+     * Sets the operator of this join configuration builder.
+     *
+     * @param operator the operator
+     * @return this builder
+     * @since 4.8.0
+     */
+    @Contract("_ -> this")
+    @NonNull Builder operator(final @NotNull UnaryOperator<Component> operator);
```

Comment:
Can you clarify that the operator does not receive the `prefix`, `suffix` or any of the `separator`s?